### PR TITLE
chore: fix workflow

### PR DESCRIPTION
**Pull request by insignias**
Issue #, if available:

Description of changes:
1. Add workflows to enable branch protection rule for `release/` branch
2. Add workflow to delete `release/` branch on merge to `main`
3. Enable and disable the branch protection rule for `develop`/`release` when committing README changes

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/release-deletion.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# This workflow disables release/* branch protection and deletes the branch after merge to develop
+name: "release deletion"
+on:
+  pull_request:
+    types: [closed]
+    branches: [main]
+
+jobs:
+  delete-release-branch-on-merge:
```
Comment (htztomic) [inline .github/workflows/release-deletion.yml:14]: can the release branch live past the merge?

Comment (insignias) [reply .github/workflows/release-deletion.yml:14]: We are merging it using the `gh api`, so we need to delete it post merge to `main`

Comment (htztomic) [reply .github/workflows/release-deletion.yml:14]: this is more of a concern if there is a need of maintaining a release post merging to main

Comment (insignias) [reply .github/workflows/release-deletion.yml:14]: Post final merge the branch should be deleted like any other branch. If we need to make fixes we can always create a new release branch from main post merge

**Review by htztomic — APPROVED** [review]

**Pull request merged**